cmd/e2e/e2e.go

```diff
@@ -63,5 +68,22 @@ func main() {
 		glog.Error("Invalid --times (negative or no testing requested)!")
 		os.Exit(1)
 	}
+
+	if context.Provider == "aws" {
+		awsConfig := "[Global]\n"
+		if cloudConfig.Zone == "" {
+			glog.Error("gce_zone must be specified for AWS")
```

Comment:
You can only have one cloud provider at a time, right?  Perhaps this is a good place for generic cloud patterns, so long as it's not planned to allow a cluster span infrastructure providers...   If that's the case, that's probably an easy small PR instead of adding more to this one.